File: typescript/libs/workspace/adapters/json/system/src/lib/runtime/Workspace.ts
import { Workspace as SystemWorkspace } from '@allors/workspace/adapters/system';
import { ISession } from '@allors/workspace/domain/system';
import { Session } from './Session';

export class Workspace extends SystemWorkspace {
  createSession(): ISession {
    return new Session(this, this.services.createSessionServices());
  }
}

File: dotnet/System/workspace/csharp/adapters/allors.workspace.adapters.remote/workspace/Workspace.cs
// <copyright file="RemoteWorkspace.cs" company="Allors bvba">
// Copyright (c) Allors bvba. All rights reserved.
// Licensed under the LGPL license. See LICENSE file in the project root for full license information.
// </copyright>

namespace Allors.Workspace.Adapters.Remote
{
    using Numbers;

    public class Workspace : Adapters.Workspace
    {
        public Workspace(DatabaseConnection database, IWorkspaceServices services, INumbers numbers, WorkspaceIdGenerator workspaceIdGenerator) : base(database, services, numbers, workspaceIdGenerator) => this.Services.OnInit(this);

        public new DatabaseConnection DatabaseConnection => (DatabaseConnection)base.DatabaseConnection;

        public override ISession CreateSession() => new Session(this, this.Services.CreateSessionServices());
    }
}
